src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs

```diff
@@ -358,6 +385,11 @@ internal virtual void Configure()
                     InitializeConstructorParameters(GetParameterInfoValues(), sourceGenMode: Options.SerializerContext != null);
                 }
             }
+
+            if (PolymorphismOptions != null)
```

Comment:
Any reference type can be polymorphic, so this includes dictionaries and collections as well. I've made it so that JsonTypeInfo surfaces the metadata even if it's applied incorrectly (e.g. on a struct). It will fail in the `Configure` stage but at least it lets the user clear or correct it on the resolver level beforehand.

Reply:
good point